### PR TITLE
solve stairs problem

### LINKED ISSUE #17: Challenge: Stairs
Solve Climbing_stairs

  
write function to count the number of ways you can climb the staircase.
add documentation
all tests are green

### PR DESCRIPTION
**Pull request by YaninaChukanova**
Challenge: Stair Climbing - Language: Python

-->

- [x] PR's title includes the challenge's name and language
- [x] PR has short and clear description of the challenge
- [x] PR has appropriate labels and milestones for easy identification
- [x] PR it is assigned to the owner
- [x] reviewers are assigned
- [x] the PR contributes only one focused change
- [x] the branch is up to date with `main`/`master`
- [x] the code works when pulled and run locally
- [x] all conflicts are resolved (if any)
- [x] It is linked to an issue in the appropriate column of the project board
- [x] feedback is addressed (if any, and if it is appropriate feedback.)

## README Documentation

- [x] The solution is documented in `/src/README.md`
- [x] The markdown source is formatted
- [x] Spelling and grammar is correct in all text
- [x] The markdown looks correct when you preview the file
- [ ] All links and images work
- [x] The README documents the solution's behavior, strategy(ies) and implementation(s)
- [x] There are use cases in code blocks to illustrate the function's behavior

## Python Files

- [x] There is a module header
- [x] There is a module docstring
- [x] File names are in snake_case
- [x] Test files are named `test_<module_name>.py`

## Function Docstring

- [x] Behavior description
- [x] Parameter description
- [x] Return value description
- [x] Include any assumptions
- [ ] Include 3 or more (passing!) doctests
- [x] Include 1-2 use cases (if necessary)

## Function Implementation

- [ ] The solution is not is not copied from [djeada](https://github.com/djeada/Algorithms-And-Data-Structures)
- [x] The code is auto-formatted
- [x] The code has no (reasonable) linting mistakes
- [x] Variables are named with snake_case
- [x] The function has a clear and helpful name
- [ ] The file's name matches the function name
- [x] The code follows the strategy as simply as possible
- [x] Variable names are clear and helpful
- [x] Comments explain the strategy (if necessary)
- [ ] There are type annotations
- [ ] (challenge) The code includes defensive assertions

## Unit Test Suite

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Each unit test has
  - [x] A helpful name
  - [ ] A clear docstring
  - [x] Only one assertion
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases
- [ ] Test suite includes black-box and glass-box tests

### CODE REVIEW

File: src/dynamic_programming/python/climbing_stairs/src/climb.py [hunk 1]
```diff
@@ -11,36 +11,35 @@ def count_ways_to_climb_basic(num_steps):
         + count_ways_to_climb_basic(num_steps - 1)
     )
```
Comment (AnnaShliapkina) [inline src/dynamic_programming/python/climbing_stairs/src/climb.py:24]: All explanation you added in README you can (and should) put in docstrings before each function. This file also need documentation.

File: src/dynamic_programming/python/climbing_stairs/src/climb.py
Comment (AnnaShliapkina) [inline src/dynamic_programming/python/climbing_stairs/src/climb.py:1]: The solution is similar to djeada solution

File: src/dynamic_programming/python/climbing_stairs/tests/test_climbing_stairs.py
Comment (AnnaShliapkina) [inline src/dynamic_programming/python/climbing_stairs/tests/test_climbing_stairs.py:1]: image

One test are not pass

**Review by AnnaShliapkina — CHANGES_REQUESTED** [review]

**Review by DmytroLyt — DISMISSED** [review]
- PR's title includes the challenge's name and language **(i didn`t find this challenge, maybe problems with naming of pull request? Also need add language)**
(Remember use PEP8 standard, code 
- There is a module header (Doesn`t have, Need to add )
- There is a module docstring (Doesn`t have, Need to add )

File climb.py need attention:
-  Behavior description (Doesn`t have, Need to add )
-  Parameter description (Doesn`t have, Need to add )
-  Return value description (Doesn`t have, Need to add )
-  The solution is not is not copied from [djeada] ( Code is looks like copied from that repo, need another solution) 


-  The test class has a docstring (Doesn`t have, Need to add )
- All tests pass ( One test failed)

**Review by AnnaShliapkina — APPROVED** [review]
